### PR TITLE
Refactor OLM reconcilation

### PR DESCRIPTION
**Pull request by csrwng**
Changes OLM reconciliation from go templates to go code
Ensures that all deployments that live on the control plane side follow
the same pattern of tolerations, taints, affinity as the rest of the
control plane components.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/olm/assets/packageserver-deployment.yaml [hunk 1]
```diff
@@ -16,11 +16,6 @@ spec:
     type: RollingUpdate
   template:
     metadata:
-      annotations:
-        include.release.openshift.io/self-managed-high-availability: "true"
-{{ if .RestartDate }}
-        openshift.io/restartedAt: "{{ .RestartDate }}"
-{{ end }}
       labels:
         app: packageserver
```
Comment (awgreene) [inline control-plane-operator/controllers/hostedcontrolplane/olm/assets/packageserver-deployment.yaml:20]: Interesting, was there a reason we decided to remove the labels from the catalog and olm deployments but not the packageserver deployment?

File: control-plane-operator/controllers/hostedcontrolplane/olm/assets/packageserver-endpoint-guest.yaml [hunk 2]
```diff
@@ -4,8 +4,7 @@ metadata:
   name: packageserver
   namespace: default
 subsets:
-- addresses:
-  - ip: {{ .PackageServerAPIClusterIP }}
```
Comment (awgreene) [inline control-plane-operator/controllers/hostedcontrolplane/olm/assets/packageserver-endpoint-guest.yaml:8]: Could you explain what this change does? Thanks!

**Review by awgreene — APPROVED** [review]
Great work @csrwng, this looks good to me. Had some non-blocking generic questions.

File: control-plane-operator/controllers/hostedcontrolplane/olm/assets/packageserver-deployment.yaml (hunk 1 above)
Comment (csrwng) [reply control-plane-operator/controllers/hostedcontrolplane/olm/assets/packageserver-deployment.yaml:20]: No reason, will remove. The reason that I removed it from the other ones is that we're now using the manifests for the spec of the resource. Nothing in the metadata is read from here.

File: control-plane-operator/controllers/hostedcontrolplane/olm/assets/packageserver-endpoint-guest.yaml (hunk 2 above)
Comment (csrwng) [reply control-plane-operator/controllers/hostedcontrolplane/olm/assets/packageserver-endpoint-guest.yaml:8]: The manifest is no longer a go template. The package server cluster IP is now set here:
https://github.com/openshift/hypershift/pull/518/files#diff-b1df563b312192d05f3b030fc9c3beac30c393525674a12c00450e77d652008bR42-R47

File: control-plane-operator/controllers/hostedcontrolplane/olm/assets/packageserver-deployment.yaml (hunk 1 above)
Comment (csrwng) [reply control-plane-operator/controllers/hostedcontrolplane/olm/assets/packageserver-deployment.yaml:20]: Actually nevermind, this is part of the deployment template. The labels are needed here for the match the deployment selector.

Comment (awgreene) [reply control-plane-operator/controllers/hostedcontrolplane/olm/assets/packageserver-deployment.yaml:20]: Ah my mistake, I thought this was the `metadata.labels` field.

**Pull request merged**